plugin/cache/cache.go

```diff
@@ -163,6 +163,9 @@ func (w *ResponseWriter) WriteMsg(res *dns.Msg) error {
 	var duration time.Duration
 	if mt == response.NameError || mt == response.NoData {
 		duration = computeTTL(msgTTL, w.minnttl, w.nttl)
+	} else if mt == response.ServerFailureError || mt == response.NotImplementedError {
+		// use default ttl which is 5s
+		duration = computeTTL(msgTTL, minTTL, maxTTL)
```

Comment:
there should not be any data in these kind of message, so we can just assume 5s as the TTL

Reply:
Done~